src/sentry/incidents/logic.py

```diff
@@ -1103,29 +1101,37 @@ def update_alert_rule_trigger_action(
         if type in AlertRuleTriggerAction.INTEGRATION_TYPES:
             integration = updated_fields.get("integration", trigger_action.integration)
             organization = trigger_action.alert_rule_trigger.alert_rule.organization
-            channel_id = get_alert_rule_trigger_action_integration_object_id(
-                type, organization, integration.id, target_identifier,
+
+            target_identifier, target_display = get_target_identifier_display_for_integration(
+                type, target_identifier, organization, integration.id,
             )
-            # Use the target identifier for display
-            updated_fields["target_display"] = target_identifier
-            updated_fields["target_identifier"] = channel_id
-        else:
+            updated_fields["target_display"] = target_display
             updated_fields["target_identifier"] = target_identifier
 
     trigger_action.update(**updated_fields)
     return trigger_action
 
 
-def get_alert_rule_trigger_action_integration_object_id(type, *args, **kwargs):
+def get_target_identifier_display_for_integration(type, target_value, *args, **kwargs):
     if type == AlertRuleTriggerAction.Type.SLACK.value:
-        return get_alert_rule_trigger_action_slack_channel_id(*args, **kwargs)
+        target_identifier = get_alert_rule_trigger_action_slack_channel_id(
+            target_value, *args, **kwargs
+        )
     elif type == AlertRuleTriggerAction.Type.MSTEAMS.value:
-        return get_alert_rule_trigger_action_msteams_channel_id(*args, **kwargs)
+        target_identifier = get_alert_rule_trigger_action_msteams_channel_id(
+            target_value, *args, **kwargs
+        )
+    elif type == AlertRuleTriggerAction.Type.PAGERDUTY.value:
+        target_value, target_identifier = get_alert_rule_trigger_action_pagerduty_service(
```

Comment:
@ceorourke Could you flip the tuple here so it returns `target_identifier`, `target_value` which is the same as what we are returning at the bottom? It's just a bit more clear that way.